komga/src/main/kotlin/org/gotson/komga/domain/model/ThumbnailSeries.kt

```diff
@@ -6,12 +6,46 @@ import java.net.URL
 import java.time.LocalDateTime
 
 data class ThumbnailSeries(
-  val url: URL,
+  val url: URL?,
   val selected: Boolean = false,
+  var thumbnail: ByteArray?,
 
   val id: String = TsidCreator.getTsid256().toString(),
   val seriesId: String = "",
 
   override val createdDate: LocalDateTime = LocalDateTime.now(),
   override val lastModifiedDate: LocalDateTime = LocalDateTime.now()
-) : Auditable(), Serializable
+) : Auditable(), Serializable {
+
+  val isCustomCover
+    get() = thumbnail != null
+
+  override fun equals(other: Any?): Boolean {
```

Comment:
Shouldn't need equals and hashcode for data classes

Reply:
IDE recommends doing it

Reply:
right, i just realized it was the case for `ThumbnailBook` 😅